pkg/pgmodel/querier/query_builder.go

```diff
@@ -164,6 +164,10 @@ func BuildSubQueries(matchers []*labels.Matcher) (*clauseBuilder, error) {
 				cb.SetMetricName(m.Value)
 				continue
 			}
+			if m.Name == pgmodel.SchemaNameLabelName {
+				cb.SetMetricName(m.Value)
+				continue
+			}
 			sq := subQueryEQ
```

Comment:
Are we not blocking people with using special `regexes` or `not` conditions on schema label name as discussed?

Reply:
Good catch, we should be throwing an error in those cases.